decomposer/solver.py

```diff
@@ -24,16 +26,11 @@ class Solver:
     # agents end their final answer on the last line after this token
     FINAL_TOKEN: str = os.getenv("FINAL_TOKEN", "vote:")
 
-    def extract_final(self, text: str, token: str = FINAL_TOKEN) -> str:
-        """
-        Return the text after the last occurrence of token (case-insensitive),
-        or the last non-empty line if not found. Preserves original casing.
-        """
-        raise NotImplementedError
```

Comment:
extract_final() no longer needs to be part of the interface, as the caller has no more use for it.